releasenotes/notes/33734.yaml

```diff
@@ -0,0 +1,19 @@
+apiVersion: release-notes/v2
+kind: feature
+
+area: traffic-management
+issue: []
+
+releaseNotes:
+- |
+  '**Improved** TCP probes now working as expected: When using TCP probes with older versions of istio the check was always successful, even if the application didn't open the port.'
+
+upgradeNotes:
+  - title: TCP probes now working as expected
+    content: |
+        When using TCP probes with older versions of istio the check was always successful, even if the application didn't open the port.
+        This may cause problems when upgrading: If you had a missconfiguration in a TCP probe (e.g wrong port) you maybe haven't noticed.
+        After the upgrade a missconfigured TCP probe will fail and therefore might cause downtimes.
```

Comment:
Can we add a feature flag to opt-out of this in https://github.com/istio/istio/blob/dbf6a9e1c123bf6abbab45683b1c6c1f7660916c/pilot/pkg/features/pilot.go#L28-L27? We can make it default on, but this allows users that may be broken by this to have more time to fix their applications.

Reply:
> Can we add a feature flag to opt-out of this?

Is it worth it? It was very strange if it worked before. IMHO, I think you even could call it `bug` that it worked before even if the port is not available. On the other hand, if we put it into the releasenotes like this, it might be good to have this option.

Reply:
@howardjohn The feature flag should be documented somewhere, right? Can you maybe point us to an example?

Reply:
Its automatically documented when you use RegisterEnvVar through some automation we have.

I totally agree with what you said, but at the same time we get users with massive deployments where even bug fixes break them because they relied on the bugs... a bit annoying, but a feature flag is pretty easy to add an makes everyone happy.